test/e2e/e2e_test.go

```diff
@@ -28,6 +37,45 @@ func init() {
 	pflag.Parse()
 }
 
+var kubebuilderBinDir string
+
+// 2 steps to do before running the e2e test suite
+// 1) go build a new kubebuilder binary
+// 2) install all required binaries, such as etcd and kube-apiserver
+// If you have run testv0.sh or testv1.sh before running this e2e suite,
+// it will build the new kubebuilder and install the required binaries.
+var _ = BeforeSuite(func(done Done) {
+	framework.TestContext.BinariesDir = "/tmp/kubebuilder/bin/"
+	// build a kubebuilder
+	targets := []string{"kubebuilder", "kubebuilder-gen"}
+	for _, target := range targets {
+		buildOptions := []string{
+			"build", "-o", path.Join(framework.TestContext.BinariesDir, target), path.Join("github.com/kubernetes-sigs/kubebuilder/cmd", target)}
+		cmd := exec.Command("go", buildOptions...)
+		cmd.Env = os.Environ()
+		command := strings.Join(cmd.Args, " ")
+		log.Printf("running %v", command)
+		output, err := cmd.CombinedOutput()
+		log.Printf("output when running:\n%s", output)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	close(done)
+}, 60)
+
+var _ = AfterSuite(func() {
+	os.RemoveAll(kubebuilderBinDir)
+	// cleanup
```

Comment:
missing any cleanup function here since you have that comment?

Reply:
Actually the cleanup is done in each test. e.g. https://github.com/kubernetes-sigs/kubebuilder/pull/310/files#diff-cc94bdab2f625f6d5f18b07a3ff121a1R39

We don't need it now :)